### PR TITLE
Upgrade React-tasks

### PR DESCRIPTION
**Pull request by NikitaBondarevv**

### CODE REVIEW

File: src/main/mount/content/index.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import styles from './styles.css';
+
+export class Content extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      counter: 0,
+      show: true,
+    };
+  }
```
Comment (iliatcymbal) [inline src/main/mount/content/index.js]: no need constructor

File: src/main/mount/content/index.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    return code;
+  }
+
+  render() {
+    const { counter, show } = this.state;
+
+    return (
+      <div className={styles.info}>
+        <button type="button" onClick={() => this.setState({ counter: counter + 1 })}>Inc</button>
```
Comment (iliatcymbal) [inline src/main/mount/content/index.js]: `() => this.setState({ counter: counter + 1 })` it should be in a separate method

File: src/main/mount/content/index.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        <span className={styles.count}>{counter}</span>
+        <br />
+        <br />
+        <button type="button" onClick={() => this.setState({ show: !show })}>
```
Comment (iliatcymbal) [inline src/main/mount/content/index.js]: `() => this.setState({ show: !show })` move it to method

File: src/main/mount/index.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { Component } from 'react';
+
+export class Mount extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      status: 'init'
+    };
```
Comment (iliatcymbal) [inline src/main/mount/index.js]: why do you need `status` state in your component?

File: src/header/nav/clock/index.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  componentWillUnmount() {
+    clearInterval(this.intervalID);
+  }
+
+  render() {
+    const { time, show } = this.state;
+
+    return (
+      <div className={styles.clock}>
+        <button type="button" className={styles.button} onClick={() => this.setState({ show: !show }, this.componentWillUnmount)}>
```
Comment (iliatcymbal) [inline src/header/nav/clock/index.js]: `this.componentWillUnmount` all life cycle methods are called by react logic inside the library core, you shouldn't call it by yourself

File: src/header/nav/clock/index.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  componentWillUnmount() {
+    clearInterval(this.intervalID);
+  }
+
+  render() {
+    const { time, show } = this.state;
+
+    return (
+      <div className={styles.clock}>
+        <button type="button" className={styles.button} onClick={() => this.setState({ show: !show })}>
```
Comment (iliatcymbal) [inline src/header/nav/clock/index.js]: using class components you shouldn't place callback right in jsx, use methods

File: src/main/mount/content/index.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+import { Mount } from '../index'
+import styles from './styles.css';
+
+export class Content extends Component {
+  state = {
+    counter: 0,
+    show: true,
+  };
+
+  toggleBackground = () => {
```
Comment (iliatcymbal) [inline src/main/mount/content/index.js]: `getBackgroundColor`

File: src/header/nav/dateNow/clock/index.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+  componentWillUnmount() {
+    clearInterval(this.intervalID);
+  }
+
+  render() {
+    const { time, show } = this.state;
+
+    return (
+      <div>
+        {show && <p className={styles.date}>{show && time.toLocaleDateString()}</p>}
```
Comment (iliatcymbal) [inline src/header/nav/dateNow/clock/index.js]: why do you need `show` here two times?

File: src/header/nav/dateNow/index.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+import styles from './styles.css';
+import { Clock } from './clock';
+
+export class DateNow extends Component {
+  state = {
+    display: true
+  }
+
+  deleteComponent = () => {
```
Comment (iliatcymbal) [inline src/header/nav/dateNow/index.js]: does it only delete component?

File: src/header/nav/dateNow/clock/index.js (hunk 8 above)
Comment (NikitaBondarevv) [reply src/header/nav/dateNow/clock/index.js:28]: You don't need twice. Error. Fixing

Comment (NikitaBondarevv) [reply src/header/nav/dateNow/clock/index.js]: No need. Error. Fixing

**Pull request merged**